product-service/handlers/get-products-by-id.js

```diff
@@ -0,0 +1,15 @@
+import { getProductsList } from "./get-products-list.js";
+
+export const getProductsById = async (event, context, callback) => {
```

Comment:
You can specify the id through the getItem method with `Key` field
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB.html#getItem-property
Fetching all products and running a loop to find the one eats more memory